lib/types/global.ts

```diff
@@ -0,0 +1,9 @@
+export type Address = {
+  host: string;
+  port: number;
+};
+
+export type XudError = {
```

Comment:
We should migrate the `error` object in the existing error classes to: `{ [key: string]: XudError }`

Reply:
I liked this idea, but as I started implementing it wound up being way more complicated given that some of our errors are not plain error objects, but rather functions that return error objects. It would have required a lot of changes and resulted in some messy code, so I'm leaving this as is.